packages/shared/src/components/Icon.tsx

```diff
@@ -37,6 +38,7 @@ const Icon = ({
   return (
     <IconComponent
       className={classNames(IconSize[size], 'pointer-events-none', className)}
+      style={size === 'large' ? { fontSize: '1.75rem' } : null}
```

Comment:
This piece i'm referring to.

What happen to medium sized buttons with a icon now, they would be not the correct size.

Also this way of adding it seems a bit hacky to me.

In my opinion we should add something in the `<Button />` component (our own one) and handle icon sizes based on the button size.

They will be different then the regular icon size in some case:
XSmall button => Small icon
Small button => Medium icon
Medium button => new w-7 h-7 map (or the large if we don't use it otherwise)
Large button => Xlarge icon

My idea is to remove the struggle for the developer to even have to think about the size, the button should handle it.
Let me know if that makes sense?

Reply:
Why medium sized buttons will have wrong icon size?

Agree that way of adding is tricky, because there is no w-7 h-7 (so we need to add it).

We can make button component handle icon size, but we still have a lot of button-looking links where we need to care about icons size